### PR TITLE
[#2858] Video no longer reports a duration or currentTime of 1hr when it...

### PR DESCRIPTION
**Pull request by humphd**
... shouldn't. Videos with a duration of over 24 hours also work fine now

### CODE REVIEW

File: src/util/lang.js [hunk 1]
```diff
@@ -68,14 +68,22 @@ define( [ "./shims" ], function(){
...
+          days,
           seconds;
+
       timeStamp.setSeconds( time );
-      seconds = timeStamp.toTimeString().substr( 0, 8 );
-      if( seconds > 86399 ){
-        seconds = Math.floor( (timeStamp - Date.parse("1/1/70") ) / 3600000) + seconds.substr(2);
+      days = timeStamp.getDay();
+
+      seconds = timeStamp.toLocaleTimeString();
```
Comment (humphd) [inline src/util/lang.js]: I think this is still liable to break.  From MDN:

"The toLocaleTimeString method relies on the underlying operating system in formatting dates. It converts the date to a string using the formatting convention of the operating system where the script is running. For example, in the United States, the month appears before the date (04/15/98), whereas in Germany the date appears before the month (15.04.98).

Methods such as getHours, getMinutes, and getSeconds give more consistent results than toLocaleTimeString. Use toLocaleTimeString when the intent is to display to the user a string formatted using the regional format chosen by the user. Be aware that this method, due to its nature, behaves differently depending on the operating system and on the user's settings.
"

I wonder if we should getHours, getMinutes, getSeconds and then concat with ':' ourselves.

File: src/util/lang.js [hunk 2]
```diff
@@ -68,14 +68,22 @@ define( [ "./shims" ], function(){
       }
     }, //smpteToSeconds
 
-    secondsToSMPTE: function( time ){
-      var timeStamp = new Date( 1970, 0, 1 ),
+    secondsToSMPTE: function( time ) {
+      var timeStamp = new Date( 0, 0, 0 ),
```
Comment (humphd) [inline src/util/lang.js]: `var timeStamp = new Date( 0 );`

Comment (dseif) [inline src/util/lang.js]: I don't think we can do that as it doesn't return what we want.

```
var a = new Date( 0 );
// returns 19 for me
a.getHours();

var b = new Date( 0, 0, 0 );
// returns 0 for me
b.getHours();
```

File: src/util/lang.js [hunk 3]
```diff
@@ -68,16 +68,26 @@ define( [ "./shims" ], function(){
...
+      timeStamp.setSeconds( +time );
+      days = timeStamp.getDay();
+      timeString = timeStamp.toTimeString().split( " " )[ 0 ];
+
+      // If the video is for some reason longer than 24 hours, we need to make sure we handle it accordingly.
+      if ( days > 0 ) {
+        (function( timeArr ) {
+          timeArr[ 0 ] = +timeArr[ 0 ] + ( days * 24 );
+          timeString = timeArr.join( ":" );
+        })( timeString.split( ":" ) );
```
Comment (secretrobotron) [inline src/util/lang.js]: There's a lint error here: `src/util/lang.js: line 86, col 11, Move the invocation into the parens that contain the function.`

While I do _love_ the creativity you employed here to introduce and utilize the `timeArr` variable, this approach is strictly more expensive than just storing `timeArr` explicitly with a  `var` statement.

This is interesting though: what happens if you have something that's longer than a day? Do the hours/minutes/seconds make sense still?

Comment (dseif) [inline src/util/lang.js]: Well considering the amount of times a user will actually have a video that is longer than 24 hours is probably very, very low, would creating a variable for `timeArr` be worth it? Seems like creating a variable for something that will happen 1 in every 1000 vidoes or something would be more of a waste than just taking this approach.

Yes, the the hours:minutes:seconds make sense when the video is over 24 hours ( shows up like 48:12:44 ).

**Pull request closed without merging**